packages/components/src/presentation-components/tree/controlled/PresentationTreeRenderer.tsx

```diff
@@ -23,6 +26,7 @@ import { useHierarchyLevelFiltering } from "./UseHierarchyLevelFiltering";
 export interface PresentationTreeRendererProps extends TreeRendererProps {
   imodel: IModelConnection;
   modelSource: TreeModelSource;
+  dataProvider: IPresentationTreeDataProvider;
```

Comment:
I'm really not a fan of passing a data provider to the renderer. Let's pass a node loader here, as discussed offline.